pxtservices/iframeEmbeddedClient.ts

```diff
@@ -0,0 +1,79 @@
+type IframeClientSetMessagePortRequest = {
+    type: "iframeclientsetmessageport";
+};
+
+type IFrameClientReadyMessage = {
+    type: "iframeclientready";
+};
+
+export type IframeClientMessage = IframeClientSetMessagePortRequest | IFrameClientReadyMessage;
+
+export class IFrameEmbeddedClient {
+    protected frameId: string;
```

Comment:
`frameId()` can return undefined. Teacher tool and other CRAs use strict typechecks and would trip on this.

```suggestion
    protected frameId: string | undefined;
```